src/main/java/org/disq_bio/disq/HtsjdkReadsRddStorage.java

```diff
@@ -35,14 +55,23 @@
     CRAM,
     /** SAM format */
     SAM;
+
+    static FormatWriteOption fromPath(String path) {
+      SamFormat samFormat = SamFormat.fromPath(path);
+      return samFormat == null ? null : samFormat.toFormatWriteOption();
+    }
   }
 
   /** An option for configuring the number of files to write a {@link HtsjdkReadsRdd} as. */
   public enum FileCardinalityWriteOption implements WriteOption {
     /** Write a single file specified by the path. */
     SINGLE,
     /** Write multiple files in a directory specified by the path. */
-    MULTIPLE
+    MULTIPLE;
```

Comment:
Here the same, it could be refactored to be:

```java
public enum FileCardinalityWriteOption implements WriteOption {
    SINGLE(opt -> opt.getSink()),
    MULTIPLE(opt -> new AnySamSinkMultiple(opt.getSamFormat()));

    // initialized on construction
    private Function<FormatWriteOption, AbstractSamSink> sinkProvider;

    ...

    public AbstractSamSink  getSink(final FormatWriteOption option) {
        return sinkProvider.apply(option);
    }
}
```